pkg/reconciler/channel/channel.go

```diff
@@ -90,13 +92,45 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, c *v1beta1.Channel) pkgr
 	}
 
 	c.Status.Channel = &backingChannelObjRef
-	c.Status.PropagateStatuses(&backingChannel.Status)
+	bCS := r.getChannelableStatus(ctx, &backingChannel.Status, backingChannel.Annotations)
+	c.Status.PropagateStatuses(bCS)
 
 	return newReconciledNormal(c.Namespace, c.Name)
 }
 
+func (r *Reconciler) getChannelableStatus(ctx context.Context, bc *duckv1alpha1.ChannelableCombinedStatus, cAnnotations map[string]string) *duckv1beta1.ChannelableStatus {
+
+	channelableStatus := &duckv1beta1.ChannelableStatus{}
+	if bc.AddressStatus.Address != nil {
+		channelableStatus.AddressStatus.Address = &duckv1.Addressable{}
+		bc.AddressStatus.Address.ConvertTo(ctx, channelableStatus.AddressStatus.Address)
+	}
+	channelableStatus.Status = bc.Status
+	if cAnnotations != nil {
+		if cAnnotations[messaging.SubscribableDuckVersionAnnotation] == "v1beta1" {
+			if len(bc.SubscribableStatus.Subscribers) > 0 {
+				channelableStatus.SubscribableStatus.Subscribers = bc.SubscribableStatus.Subscribers
+			}
+		} else { //v1alpha1
+			if bc.SubscribableTypeStatus.SubscribableStatus != nil &&
+				len(bc.SubscribableTypeStatus.SubscribableStatus.Subscribers) > 0 {
+				channelableStatus.SubscribableStatus.Subscribers = make([]duckv1beta1.SubscriberStatus, len(bc.SubscribableTypeStatus.SubscribableStatus.Subscribers))
+				for i, ss := range bc.SubscribableTypeStatus.SubscribableStatus.Subscribers {
+					channelableStatus.SubscribableStatus.Subscribers[i] = duckv1beta1.SubscriberStatus{
+						UID:                ss.UID,
+						ObservedGeneration: ss.ObservedGeneration,
+						Ready:              ss.Ready,
+						Message:            ss.Message,
+					}
+				}
+			}
+		}
+	}
+	return channelableStatus
+}
+
 // reconcileBackingChannel reconciles Channel's 'c' underlying CRD channel.
-func (r *Reconciler) reconcileBackingChannel(ctx context.Context, channelResourceInterface dynamic.ResourceInterface, c *v1beta1.Channel, backingChannelObjRef duckv1.KReference) (*duckv1beta1.Channelable, error) {
+func (r *Reconciler) reconcileBackingChannel(ctx context.Context, channelResourceInterface dynamic.ResourceInterface, c *v1beta1.Channel, backingChannelObjRef duckv1.KReference) (*duckv1alpha1.ChannelableCombined, error) {
 	lister, err := r.channelableTracker.ListerForKReference(backingChannelObjRef)
```

Comment:
Did you also change the channelableTracker in the controller?

Reply:
Something like this:
https://github.com/knative/eventing/blob/master/pkg/reconciler/subscription/controller.go#L59

Reply:
thanks!!